geoh5py/objects/curve.py

```diff
@@ -155,6 +169,24 @@ def parts(self, indices: list | np.ndarray):
             self._cells = None
             self.workspace.update_attribute(self, "cells")
 
+    def remove_cells(self, indices: list[int]):
+        """Safely remove cells and corresponding data entries."""
+
+        if self._cells is None:
+            raise UserWarning("No cells to be removed.")
```

Comment:
Should this be an AttributeError? If you want to warn, better to use the warning module